// @flow
import test from '../../index.js.flow';

test('test', async t => {
	t.throws(() => { throw new Error(); });
	await t.throwsAsync(Promise.reject(new Error()));

	t.notThrows(() => { return; });
	await t.notThrowsAsync(Promise.resolve('Success'));

	const error = t.throws(() => { throw new Error(); });
	const message: string = error.message;

	const errorAsync = await t.throwsAsync(Promise.reject(new Error()));
	const messageAsync: string = errorAsync.message;
});
